### PR TITLE
Add an Image Fetcher Example

### PR DESCRIPTION
**Pull request by DavidMansolino**
see https://github.com/aiwc/test_world/issues/58#issuecomment-520336505

### CODE REVIEW

File: controllers/supervisor/image_frame_buffer.py [hunk 1]
```diff
@@ -44,9 +44,10 @@ def update_image(self, time):
...
                         break
                 if changed:
-                    b64_encoded = ''
+                    pixels = []
                     for py in range(yStart, yEnd):
                         index = 4 * (py * self.width + xStart)
-                        b64_encoded += base64.b64encode(bytes(self.currentImage[index:index + xLength * 4])).decode("utf-8")
-                    ret.append([xStart, yStart, xLength, yLength, b64_encoded])
+                        pixels.extend(self.currentImage[index:index + xLength * 4])
+                    ret.append([xStart, yStart, xLength, yLength, base64.standard_b64encode(bytes(pixels))])
```
Comment (DavidMansolino) [inline controllers/supervisor/image_frame_buffer.py]: The base64 encoding is sometimes returning strange values such as:
> ...'\\x00', '\\xff', **'{'**, **'{'**, **'{'**, '\\xff', '\\x00', '\\xd1'...

Not sure to understand where does these `{` come from

**Comment by DavidMansolino** [comment]
Almost :smile: 

Capture du 2019-08-13 08-56-47

**Comment by DavidMansolino** [comment]
:tada: 

Capture du 2019-08-13 10-59-57

As expected, the actual fix was a one-line detail:  1a99ffb36118f552f457c0a1f7761f43a1d7f763

**Comment by DavidMansolino** [comment]
:+1:

**Comment by omichel** [comment]
:eyes:

File: controllers/supervisor/image_frame_buffer.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
```
Comment (omichel) [inline controllers/supervisor/image_frame_buffer.py]:
```suggestion
#!/usr/bin/env python3
```

File: controllers/supervisor/image_frame_buffer.py [hunk 3]
```diff
@@ -44,9 +44,9 @@ def update_image(self, time):
...
                         break
                 if changed:
-                    b64_encoded = ''
+                    pixels = []
                     for py in range(yStart, yEnd):
                         index = 4 * (py * self.width + xStart)
-                        b64_encoded += base64.b64encode(bytes(self.currentImage[index:index + xLength * 4])).decode("utf-8")
-                    ret.append([xStart, yStart, xLength, yLength, b64_encoded])
+                        pixels.extend(self.currentImage[index:index + xLength * 4])
+                    ret.append([xStart, yStart, xLength, yLength, base64.b64encode(bytes(pixels)).decode("utf-8")])
```
Comment (omichel) [inline controllers/supervisor/image_frame_buffer.py]:
```suggestion
                    ret.append([xStart, yStart, xLength, yLength, base64.b64encode(bytes(pixels)).decode()])
```
utf-8 is the default encode value.

File: examples_new/general_image-fetch_py/general_image-fetch.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# Maintainer: Chansol Hong (cshong@rit.kaist.ac.kr)
+
+import base64
+import numpy as np
+from PIL import Image
+import os
+import sys
+sys.path.append(os.path.dirname(os.path.realpath(__file__)) + '/../common')
+try:
+    print(sys.path)
```
Comment (omichel) [inline examples_new/general_image-fetch_py/general_image-fetch.py]: Is that a debug message?

```suggestion
```

File: controllers/supervisor/supervisor.py [hunk 5]
```diff
@@ -214,6 +214,8 @@ def set_speeds(self, team, speeds):
                 )
 
     def callback(self, client, message):
+        if not message.startswith('aiwc.') and 'aiwc.' in message:
+            message = 'aiwc.' + message.split('aiwc.', 1)[-1]
```
Comment (omichel) [inline controllers/supervisor/supervisor.py]: That seems dangerous to trash part of the message... Instead, we should ensure that this part of the message was processed before (and thus we could remove these two lines).

**Review by omichel — CHANGES_REQUESTED** [review]

File: controllers/supervisor/image_frame_buffer.py (hunk 2 above)
Comment (DavidMansolino) [reply controllers/supervisor/image_frame_buffer.py]: That was to be homogeneous with the previous version of the controllers: https://github.com/aiwc/examples/blob/ae937e69e40b82159e21861d86852c5ae4425321/commentator_skeleton_py/commentator_skeleton.py#L1

But your version seems indeed more correct.

**Comment by DavidMansolino** [comment]
:+1:

File: examples_new/general_image-fetch_py/general_image-fetch.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python3
```
Comment (omichel) [inline examples_new/general_image-fetch_py/general_image-fetch.py]:
```suggestion
#!/usr/bin/env python3
```

**Comment by DavidMansolino** [comment]
:+1:

**Review by omichel — APPROVED** [review]
Thank you!

**Pull request merged**